domain/src/test/java/org/oppia/android/domain/oppialogger/analytics/CpuPerformanceSnapshotterTest.kt

```diff
@@ -0,0 +1,352 @@
+class CpuPerformanceSnapshotterTest {
+
+  @Inject lateinit var cpuPerformanceSnapshotter: CpuPerformanceSnapshotter
+  @Inject lateinit var fakePerformanceMetricsEventLogger: FakePerformanceMetricsEventLogger
+  @Inject lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+  @Inject lateinit var applicationLifecycleObserver: ApplicationLifecycleObserver
+  @Inject lateinit var fakePerformanceMetricAssessor: FakePerformanceMetricAssessor
+  @Inject lateinit var fakeOppiaClock: FakeOppiaClock
+
+  @field:[JvmField Inject ForegroundCpuLoggingTimePeriodMillis]
+  var foregroundCpuLoggingTimePeriodMillis: Long = Long.MIN_VALUE
+
+  @field:[JvmField Inject BackgroundCpuLoggingTimePeriodMillis]
+  var backgroundCpuLoggingTimePeriodMillis: Long = Long.MIN_VALUE
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testSnapshotter_updateIconificationToBackground_logsCpuUsageInBackgroundState() {
+    fakePerformanceMetricAssessor.setRelativeCpuUsage(TEST_CPU_USAGE_ONE)
+    cpuPerformanceSnapshotter.updateAppIconification(APP_IN_BACKGROUND)
+    testCoroutineDispatchers.runCurrent()
+    testCoroutineDispatchers.advanceTimeBy(backgroundCpuLoggingTimePeriodMillis)
+
+    val event = fakePerformanceMetricsEventLogger.getMostRecentPerformanceMetricsEvent()
+
+    assertThat(event.loggableMetric.loggableMetricTypeCase).isEqualTo(CPU_USAGE_METRIC)
+    assertThat(event.loggableMetric.cpuUsageMetric.cpuUsageMetric).isEqualTo(TEST_CPU_USAGE_ONE)
+    assertThat(event.currentScreen).isEqualTo(ScreenName.BACKGROUND_SCREEN)
+  }
+
+  @Test
+  fun testSnapshotter_updateIconificationToForeground_logsCpuUsageInForegroundState() {
+    fakePerformanceMetricAssessor.setRelativeCpuUsage(TEST_CPU_USAGE_ONE)
+    cpuPerformanceSnapshotter.updateAppIconification(APP_IN_FOREGROUND)
+    testCoroutineDispatchers.runCurrent()
+    testCoroutineDispatchers.advanceTimeBy(foregroundCpuLoggingTimePeriodMillis)
+
+    val event = fakePerformanceMetricsEventLogger.getMostRecentPerformanceMetricsEvent()
+
+    assertThat(event.loggableMetric.loggableMetricTypeCase).isEqualTo(CPU_USAGE_METRIC)
+    assertThat(event.loggableMetric.cpuUsageMetric.cpuUsageMetric).isEqualTo(TEST_CPU_USAGE_ONE)
+    assertThat(event.currentScreen).isEqualTo(ScreenName.FOREGROUND_SCREEN)
+  }
+
+  @Test
+  fun testSnapshotter_moveToForeground_moveToBackground_verifySequentialLogging() {
+    cpuPerformanceSnapshotter.updateAppIconification(APP_IN_FOREGROUND)
+    testCoroutineDispatchers.runCurrent()
+    cpuPerformanceSnapshotter.updateAppIconification(APP_IN_BACKGROUND)
+    testCoroutineDispatchers.runCurrent()
+    testCoroutineDispatchers.advanceTimeBy(backgroundCpuLoggingTimePeriodMillis)
+
+    val count = fakePerformanceMetricsEventLogger.getPerformanceMetricsEventListCount()
+    val latestEvents =
+      fakePerformanceMetricsEventLogger.getMostRecentPerformanceMetricsEvents(count)
+    // event that got logged after time advancement.
```

Comment:
```suggestion
    // Event that got logged after time advancement.
```

Per convention. Ditto everywhere else in this PR.

Reply:
done.